### PR TITLE
Feature/search n noti page frontend

### PR DESCRIPTION
**Pull request by jlmjkfd**
The notification page is waiting to merge with the sidebar in another branch.
Add a container to layout with nav
Add some blue colors (different brightness of primary color) in tailwind config

### CODE REVIEW

**Review by xinranwangxwan559 — APPROVED** [review]

File: frontend/src/pages/SearchPage.jsx [hunk 1]
```diff
@@ -1,3 +1,292 @@
...
+  return (
+    <>
+      {/* search box */}
+      <div className="mt-10 flex space-x-4 w-2/3 m-auto border-2 border-primary rounded-full pl-4">
+        <input
+          id="Search"
+          className="flex-grow text-2xl outline-none pl-4"
+          type="text"
+        />
+        <Button className="block" onClick={handleSearchBthClick}>
```
Comment (ryanwooa) [inline frontend/src/pages/SearchPage.jsx]: How about have some gap between button and the outline of input.

File: frontend/src/pages/SearchPage.jsx [hunk 2]
```diff
@@ -1,3 +1,292 @@
...
+    mykeywords = document.getElementById("Search").value;
+    setDisplayedGroups(fetchedData.filter((group) => group.type === "Course"));
+    setKeywordList(splitKeywords(mykeywords));
+  }
+
+  return (
+    <>
+      {/* search box */}
+      <div className="mt-10 flex space-x-4 w-2/3 m-auto border-2 border-primary rounded-full pl-4">
+        <input
```
Comment (ryanwooa) [inline frontend/src/pages/SearchPage.jsx]: How about add a search Icon at the beginning of search input

File: frontend/src/pages/SearchPage.jsx [hunk 3]
```diff
@@ -1,3 +1,292 @@
...
+    const regstr = kwords.join("|");
+    const regex = new RegExp(`(${regstr})`, "gi");
+    return text.replace(
+      regex,
+      '<span style="background-color: yellow">$1</span>'
+    );
+  }
+
+  return (
+    <div className="flex flex-col border-2 border-hmblue-100 m-4 rounded-lg px-4 py-2">
```
Comment (ryanwooa) [inline frontend/src/pages/SearchPage.jsx:145]: How about have gradient border color

File: frontend/src/pages/SearchPage.jsx [hunk 4]
```diff
@@ -1,3 +1,292 @@
...
+            __html: highlightKeywords(group.title, keywords),
+          }}
+        ></div>
+        <div className="font-thin text-gray-400 text-sm">
+          {`Members: ${group.currentMember}/[${group.minMember}-${group.maxMember}]`}
+        </div>
+      </div>
+      <div className="flex justify-start space-x-1 text-xs mt-2 overflow-hidden">
+        {group.tags.map((tag, idx) => (
+          <span
```
Comment (ryanwooa) [inline frontend/src/pages/SearchPage.jsx:166]: How about different color?

**Review by ryanwooa — APPROVED** [review]

**Comment by tpan501** [comment]
image
the search bar is covered by the navbar in my browser

**Comment by jlmjkfd** [comment]
> image the search bar is covered by the navbar in my browser

Yes, waiting for Ryan to update the nav bar.

**Review by zli213 — APPROVED** [review]
I have three suggestions:
1. Add a project activity option.
2. Unify the style of label corner rounding.
3. The border color is a gradient in Figma; we should discuss whether to implement this detail.
CleanShot 2024-04-14 at 14 41 40@2x

**Review by ChelseaHuang748 — APPROVED** [review]

**Pull request merged**